core/views.py

```diff
@@ -103,6 +103,31 @@ def post(self: "PostVoteView", request: HttpRequest, pk: int, vote_type: str) ->
         return redirect("post-detail", pk=post.id)
 
 
+class PostAwardCreateView(LoginRequiredMixin, CreateView):
+    model = PostAward
+    form_class = PostAwardForm
+    template_name = "core/post-award.html"
+
+    def get_context_data(self: "PostAwardCreateView", **kwargs: any) -> dict[str, any]:
+        context = super().get_context_data(**kwargs)
+        post = get_object_or_404(Post, pk=self.kwargs["pk"])
+        context["post"] = post
+        return context
+
+    def form_valid(self: "PostAwardCreateView", form: PostAwardForm) -> HttpResponseRedirect:
+        form.instance.user = self.request.user  # Set current user
+        form.instance.post = get_object_or_404(Post, pk=self.kwargs["pk"])  # Get post with pk from URL
+
+        try:
+            return super().form_valid(form)
+        except DuplicateAwardError as e:
```

Comment:
Ah, here is catching the exception. Good one! Disregard my comment on that exception in models.py

Reply:
reorganized. Done.